components/cloud/gcp/src/gcs.rs

```diff
@@ -345,6 +395,82 @@ impl BlobStorage for GcsStorage {
     }
 }
 
+struct GcsPrefixIter<'cli> {
+    cli: &'cli GcsStorage,
+    page_token: Option<String>,
+    prefix: String,
+    finished: bool,
+}
+
+impl<'cli> GcsPrefixIter<'cli> {
+    async fn one_page(&mut self) -> io::Result<Option<Vec<BlobObject>>> {
```

Comment:
Is there any difference about function name between `s3::next_page` and `gcs::one_page`?